src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc

```diff
@@ -1333,9 +1333,10 @@ void GrpcLb::ProcessChannelArgsLocked(const grpc_channel_args& args) {
 
 void GrpcLb::UpdateLocked(const grpc_channel_args& args) {
   ProcessChannelArgsLocked(args);
-  // If fallback is configured and the RR policy already exists, update
-  // it with the new fallback addresses.
-  if (lb_fallback_timeout_ms_ > 0 && rr_policy_ != nullptr) {
+  // If we are in fallback mode (i.e., we haven't received any serverlist yet,
+  // but we have an RR policy), update the RR policy with the new fallback
+  // addresses.
+  if (serverlist_ == nullptr && rr_policy_ != nullptr) {
```

Comment:
The resolver might have changed channel args other than the fallback addresses, so I think we do want to update the RR policy even if the serverlist is set.  The code in `CreateOrUpdateRoundRobinPolicyLocked()` will do the right thing either way.

Also, if fallback mode is disabled (i.e., `lb_fallback_timeout_ms_` is 0), then we don't want to create the RR policy here.

I think the code is correct as-is, without this PR.

Reply:
Yes, the update includes more than just the fallback addresses. 

If fallback mode is disabled and the RR policy is created from the serverlist, we should still update the RR policy, right? So I think the right thing to do is to update the RR policy with the new channel args as long as we have an RR policy.